### PR TITLE
Remove the Http Client Merger mechanism

### LINKED ISSUE #554: Remove the config merger strategy
Prism right now has a merging strategy for both configurations and components (mocker, forwarder) in name of a flexibility we do not need.

Let's remove all this code we do not need so that the readability improves and we can move forward to expose a good HTTP Client API

### PR DESCRIPTION
**Pull request by XVincentX**
The following PR is a mid-size refactor that's going to be needed for the new HTTP Client API.

Hopefully this code removal will simplify a lot of the complicated process to create the config object which had load of features nobody was using and nobody will use.

*Note:* The code was so articulated it took me two days to figure it out what was going on.

In particular:

* Removal of the Prism's factory of the factory (which was used to change the configurations and the components on too many places). This was not used anywhere.
* Removed the weird API that was allowing to do stuff like `router.route(data, alternativeRouter)`
* Removal of the `IPrismConfigurationProvider` function concept that was used to get the informations from the HTTP Request — we're using a simpler approach which is just baked in the http-server — that is where the logic should really be
* Removal of all tests and stuff we're not using anymore
* Removal of the forwarder logic that was not used anywhere and was not even configurable

Closes #554

### CODE REVIEW

File: packages/core/src/factory.ts [hunk 1]
```diff
@@ -3,165 +3,139 @@ import * as Either from 'fp-ts/lib/Either';
...
+                    },
                     config: configObj,
+                  })(components.logger.child({ name: 'NEGOTIATOR' })),
+                ),
+                TaskEither.map(output => ({ output, resource })),
+              );
+            } else if (components.forwarder) {
+              // forward request and set output from response
+              return pipe(
+                components.forwarder.fforward({
```
Comment (philsturgeon) [inline packages/core/src/factory.ts]: Is this forwarder logic used somewhere? We’re meant to be removing the forwarder as it exists in favour of clean and obvious separation of HTTP server and HTTP client. Anything that’s forwarder related which is not directly being used to power the HTTP Client should be removed and/or refactored as we go. Doesn’t have to be done in this PR though.

File: packages/http-server/src/server.ts [hunk 2]
```diff
@@ -52,11 +53,12 @@ export const createServer = (operations: IHttpOperation[], opts: IPrismHttpServe
     return done(undefined, payload);
   });
 
-  const mergedConfig = configMergerFactory(
-    { cors: false, mock: { dynamic: false }, validateRequest: true, validateResponse: true },
-    config,
-    getHttpConfigFromRequest,
-  );
+  const mergedConfig = defaults<Partial<IHttpConfig>, IHttpConfig>(config, {
+    cors: true,
```
Comment (philsturgeon) [inline packages/http-server/src/server.ts]: Are we intentionally changing cors from false to true here?

File: packages/http/src/mocker/HttpMocker.ts [hunk 3]
```diff
@@ -168,3 +168,5 @@ function computeBody(
   }
   return undefined;
 }
+
+export const mocker = new HttpMocker();
```
Comment (philsturgeon) [inline packages/http/src/mocker/HttpMocker.ts]: Do we want a singleton here? Is there any weird state stuff which can happen with this class?

Comment (XVincentX) [reply packages/http/src/mocker/HttpMocker.ts]: All the other components were singletons as well — I've just unified the code. The mocker is stateless, so singleton does not change the game. Ultimately I want to remove this as a class — the mocker should be nothing more than a function. Baby steps!

File: packages/http-server/src/server.ts (hunk 2 above)
Comment (XVincentX) [reply packages/http-server/src/server.ts]: Yes! `cors` is enabled by default but the whole merger mechanism was hiding this automatically injecting stuff from 300 different places and functions executed here and there. By removing this and providing an explicit path, these errors popped up quickly and have been solved.

File: packages/core/src/factory.ts (hunk 1 above)
Comment (XVincentX) [reply packages/core/src/factory.ts]: Yes, I'm actually super ok removing the whole part and it should be really easy.

**Comment by philsturgeon** [comment]
Great feedback and love the direction this is going in. Let me know when you want a review!

File: packages/http-server/src/server.ts (hunk 2 above)
Comment (XVincentX) [reply packages/http-server/src/server.ts]: Done in c4cfd24fcaac0e58a92bda907d519e5c1bf77064

File: packages/core/src/factory.ts (hunk 1 above)
Comment (XVincentX) [reply packages/core/src/factory.ts]: Done in c4cfd24fcaac0e58a92bda907d519e5c1bf77064

**Pull request merged**